instaclustr/resource_cluster.go

```diff
@@ -637,66 +816,82 @@ func resourceClusterRead(d *schema.ResourceData, meta interface{}) error {
+		if err != nil {
+			return err
+		}
+		if err := d.Set("bundle", bundles); err != nil {
+			return fmt.Errorf("[Error] Error reading cluster: %s", err)
+		}
 	}
 
-	nodeSize := ""
-	/*
-	*  Ideally, we would like this information to be coming directly from the API cluster status.
-	*  Hence, this is a slightly hacky way of ignoring zookeeper node sizes (Kafka bundles specific).
-	 */
-	for _, node := range cluster.DataCentres[0].Nodes {
-		nodeSize = node.Size
-		if !strings.HasPrefix(nodeSize, "zk-") {
-			break
+	if isClusterSingleDataCentre(*cluster) {
+		clusterProvider := make(map[string]interface{}, 0)
+		mapstructure.Decode(cluster.Provider[0], &clusterProvider)
+		processedClusterProvider := processProvider(d, clusterProvider)
+		d.Set("cluster_provider", processedClusterProvider)
+
+		nodeSize := ""
+		/*
+		*  Ideally, we would like this information to be coming directly from the API cluster status.
+		*  Hence, this is a slightly hacky way of ignoring zookeeper node sizes (Kafka bundles specific).
+		 */
+		for _, node := range cluster.DataCentres[0].Nodes {
+			nodeSize = node.Size
+			if !strings.HasPrefix(nodeSize, "zk-") {
+				break
+			}
 		}
-	}
 
-	//loop over each data centre to determine nodes and racks for rack allocation
-	nodeCount := 0
-	rackList := make([]string, 0)
-	for _, dataCentre := range cluster.DataCentres {
-		for _, node := range dataCentre.Nodes {
+		//loop over each data centre to determine nodes and racks for rack allocation
+		nodeCount := 0
+		rackCount := 0
+		rackList := make([]string, 0)
+		for _, node := range cluster.DataCentres[0].Nodes {
 			if !strings.HasPrefix(node.Size, "zk-") {
 				nodeCount += 1
 			}
 			rackList = appendIfMissing(rackList, node.Rack)
 		}
-	}
-	rackCount := len(rackList)
-	nodesPerRack := nodeCount / rackCount
+		rackCount += len(rackList)
 
-	rackAllocation := make(map[string]interface{}, 0)
-	rackAllocation["number_of_racks"] = strconv.Itoa(rackCount)
-	rackAllocation["nodes_per_rack"] = strconv.Itoa(nodesPerRack)
+		nodesPerRack := nodeCount / rackCount
+		rackAllocation := make(map[string]interface{}, 0)
+		rackAllocation["number_of_racks"] = strconv.Itoa(rackCount)
+		rackAllocation["nodes_per_rack"] = strconv.Itoa(nodesPerRack)
 
-	if err := d.Set("rack_allocation", rackAllocation); err != nil {
-		return fmt.Errorf("[Error] Error reading cluster, rack allocation could not be derived: %s", err)
-	}
-	if len(cluster.DataCentres[0].ResizeTargetNodeSize) > 0 {
-		nodeSize = cluster.DataCentres[0].ResizeTargetNodeSize
+		if err := d.Set("rack_allocation", rackAllocation); err != nil {
+			return fmt.Errorf("[Error] Error reading cluster, rack allocation could not be derived: %s", err)
+		}
+
+		if len(cluster.DataCentres[0].ResizeTargetNodeSize) > 0 {
+			nodeSize = cluster.DataCentres[0].ResizeTargetNodeSize
+		}
+
+		d.Set("node_size", nodeSize)
+		d.Set("data_centre", cluster.DataCentres[0].Name)
+		d.Set("cluster_network", cluster.DataCentres[0].CdcNetwork)
```

Comment:
[question] we seem to be referring to DataCentres[0] here but isn't this for a single DC cluster which wont have any `DataCentres` only `DataCentre`? Same with line 851 above